src/GraphControl/Control/Grapher.cpp

```diff
@@ -433,6 +439,105 @@ namespace GraphControl
+            {
+                return graph;
+            }
+        }
+
+        return nullptr;
+    }
+
+    void Grapher::UpdateKeyGraphFeatures()
+    {
+        auto equations = GetValidEquations();
+        for (auto equation : GetValidEquations())
+        {
+            if (auto graph = GetGraph(equation))
+            {
+                if (auto analyzer = graph->GetAnalyzer())
+                {
+                    if (analyzer->CanFunctionAnalysisBePerformed())
+                    {
+                        m_solver->FormatOptions().SetFormatType(FormatType::MathML);
+                        if (S_OK
+                            == analyzer->PerformFunctionAnalysis(
+                                (Graphing::Analyzer::NativeAnalysisType)Graphing::Analyzer::PerformAnalysisType::PerformAnalysisType_All))
+                        {
+                            Graphing::IGraphFunctionAnalysisData functionAnalysisData = m_solver->Analyze(analyzer.get());
+                            {
+                                equation->XIntercept = ref new String(functionAnalysisData.Zeros.c_str());
+                                equation->YIntercept = ref new String(functionAnalysisData.YIntercept.c_str());
+                                equation->Domain = ref new String(functionAnalysisData.Domain.c_str());
+                                equation->Range = ref new String(functionAnalysisData.Range.c_str());
+                                equation->Parity = functionAnalysisData.Parity;
+
+                                // There is only ever one item in Periodicity. Map is being used because it works with DependencyProperty.
+                                for (auto p : functionAnalysisData.Periodicity)
+                                {
+                                    equation->Periodicity->Insert(p.first.ToString(), ref new String(p.second.c_str()));
+                                }
+
+                                equation->Minima = ConvertWStringVector(functionAnalysisData.Minima);
+                                equation->Maxima = ConvertWStringVector(functionAnalysisData.Maxima);
+                                equation->InflectionPoints = ConvertWStringVector(functionAnalysisData.InflectionPoints);
+                                equation->Monotonicity = ConvertWStringIntMap(functionAnalysisData.MonotoneIntervals);
+                                equation->VerticalAsymptotes = ConvertWStringVector(functionAnalysisData.VerticalAsymptotes);
+                                equation->HorizontalAsymptotes = ConvertWStringVector(functionAnalysisData.HorizontalAsymptotes);
+                                equation->ObliqueAsymptotes = ConvertWStringVector(functionAnalysisData.ObliqueAsymptotes);
+                                equation->TooComplexFeatures = functionAnalysisData.TooComplexFeatures;
+                                equation->AnalysisError = (int)AnalysisErrorType::NoError;
```

Comment:
```suggestion
                                equation->AnalysisError = static_cast<int>(AnalysisErrorType::NoError);
```

Use static_cast/dynamic_cast instead here and in the rest of the code